### ISSUE TITLE
runtime.ClientRequest and multiple values for a parameter usage

### ISSUE THREAD

**Issue by elakito** [go-openapi/runtime#68]
Parameters such as query or form parameters allow multiple values per key and there are corresponding API in most web frameworks to retrieve those values, as in here
https://docs.oracle.com/javaee/7/api/javax/servlet/ServletRequest.html#getParameter-java.lang.String-
https://docs.oracle.com/javaee/7/api/javax/servlet/ServletRequest.html#getParameterValues-java.lang.String-

I observe below that runtime.ClientRequest's SetHeaderParam, SetQueryParam, and SetFormParam methods allow multiple values to be set.
https://github.com/go-openapi/runtime/blob/master/client_request.go#L39-L53

A query parameter with multiple values is correctly processed, but a form parameter seems to use only the first entry from the below code.
https://github.com/go-openapi/runtime/blob/master/client/request.go#L119-L126

So, my first question is, shouldn't we be handling multiple values here?

Another question is that SetFileParam takes a single value instead of taking optionally multiple values as in the other methods. Here, shouldn't we have a multivalue signature as in the other methdos?

I have a target service which expects multiple values for a file parameter, in other words, we expect multiple parts for the same name, as mentioned here.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition#As_a_header_for_a_multipart_body

I would appreciate if you can provide me the answer to my questions.
regards, aki

**Comment by casualjim** [comment]
you're right about the files but I think an openapi 2.0 spec also doesn't allow for doing multiple files in the same form field. At least I can't find anything about it in the specification.

The multiple values for the form parameters are indeed an issue, but only when the collectionformat is multi

**Comment by elakito** [comment]
Hi @casualjim, regarding the missing support of multiple files from the spec perspective, I saw this issue
https://github.com/OAI/OpenAPI-Specification/issues/254

So, I wanted to to workaround this issue by customizing the generated code (basically customizing he part calling the openapi code) to be able to call our legacy (code-first) service that has this multi-file behavior. However, this was not possible because of the openapi library didn't support multi form/file parameters.

Would it be possible to support this in openapi library to make this possible?

regards, aki

**Comment by elakito** [comment]
hi,
I have a proposed fix for supporting multiple files per file-param https://github.com/elakito/runtime/commit/a9380a421910f27916b82df0ea044cff8da0c14d

I would appreciate for your comments.
regards, aki

**Comment by casualjim** [comment]
That looks fine to me. This should be a very straightforward fix for people. 💯

**Comment by elakito** [comment]
@casualjim thanks. I sent a PR.
regards, aki
